### PR TITLE
feature: support custom service name when registering with nacos

### LINKED ISSUE #2355: feature: Set configurable service name automatically when registering with nacos
## Why you need it?
希望 seata-server 在向nacos注册时自动设置可配置的服务名，目前的seata在nacos中的服务名为serverAddr不利于理解。

## Other related information
image

### PR DESCRIPTION
**Pull request by CvShrimp**
… with nacos

### Ⅰ. Describe what this PR did
Set configurable service name automatically when registering with nacos

### Ⅱ. Does this pull request fix one issue?

fixes #2355

### CODE REVIEW

File: discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java [hunk 1]
```diff
@@ -203,6 +205,14 @@ private static String getClusterName() {
         return cluster;
     }
 
+    private static String getServiceName() {
+        String serviceName = FILE_CONFIG.getConfig(getNacosApplicationFileKey());
```
Comment (slievrly) [inline discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java]: getConfig support defaultValue

File: discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java [hunk 2]
```diff
@@ -43,10 +43,12 @@
 public class NacosRegistryServiceImpl implements RegistryService<EventListener> {
     private static final String DEFAULT_NAMESPACE = "";
     private static final String DEFAULT_CLUSTER = "default";
+    private static final String DEFAULT_APPLICATION = "seata";
     private static final String PRO_SERVER_ADDR_KEY = "serverAddr";
```
Comment (slievrly) [inline discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java]: may be can remove unused.

File: discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java [hunk 3]
```diff
@@ -43,10 +43,12 @@
 public class NacosRegistryServiceImpl implements RegistryService<EventListener> {
     private static final String DEFAULT_NAMESPACE = "";
     private static final String DEFAULT_CLUSTER = "default";
+    private static final String DEFAULT_APPLICATION = "seata";
```
Comment (slievrly) [inline discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java]: how about `seata-server` ?

File: server/src/main/resources/registry.conf [hunk 4]
```diff
@@ -3,6 +3,7 @@ registry {
   type = "file"
 
   nacos {
+    application = "seata"
```
Comment (slievrly) [inline server/src/main/resources/registry.conf]: where your client configuration item `application`

File: discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java (hunk 3 above)
Comment (CvShrimp) [reply discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java]: ok，i will optimize it

File: discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java (hunk 1 above)
Comment (CvShrimp) [reply discovery/seata-discovery-nacos/src/main/java/io/seata/discovery/registry/nacos/NacosRegistryServiceImpl.java]: ok，i have already optimize it

**Comment by slievrly** [comment]
@CvShrimp https://github.com/seata/seata/tree/develop/script/client/spring

**Comment by CvShrimp** [comment]
> @CvShrimp https://github.com/seata/seata/tree/develop/script/client/spring

Thx, i have already add the client configuration

**Comment by a364176773** [comment]
please resolve code conflicts

**Comment by CvShrimp** [comment]
> please resolve code conflicts

i have already resolve the code conflicts

**Pull request merged**